src/spikeinterface/curation/curation_model.py

```diff
@@ -0,0 +1,360 @@
+                if labels is None:
+                    labels = set(manual_label.keys()) - {"unit_id"}
+                    manual_label["labels"] = {}
+                else:
+                    manual_label["labels"] = {k: list(v) for k, v in labels.items()}
+                for label in labels:
+                    if label not in values["label_definitions"]:
+                        raise ValueError(f"Manual label {unit_id} has an unknown label {label}")
+                    if label not in manual_label["labels"]:
+                        if label in manual_label:
+                            manual_label["labels"][label] = list(manual_label[label])
+                        else:
+                            raise ValueError(f"Manual label {unit_id} has no value for label {label}")
+                if unit_id not in unit_ids:
+                    raise ValueError(f"Manual label unit_id {unit_id} is not in the unit list")
+                manual_labels[i] = manual_label
+            values["manual_labels"] = manual_labels
+        return values
+
+    @classmethod
+    def check_merges(cls, values):
+
+        unit_ids = list(values["unit_ids"])
+        merges = values.get("merges")
+        if merges is None:
+            values["merges"] = []
+            return values
+
+        if isinstance(merges, dict):
+            # Convert dict format to list of Merge objects
+            merge_list = []
+            for merge_new_id, merge_group in merges.items():
+                merge_list.append({"merge_unit_group": list(merge_group), "merge_new_unit_id": merge_new_id})
+            merges = merge_list
+
+        # Make a copy of the list
+        merges = list(merges)
+
+        # Convert items to Merge objects
+        for i, merge in enumerate(merges):
+            if isinstance(merge, list):
+                merge = {"merge_unit_group": list(merge)}
+            if isinstance(merge, dict):
+                merge = dict(merge)
+                if "merge_unit_group" in merge:
+                    merge["merge_unit_group"] = list(merge["merge_unit_group"])
+                merges[i] = Merge(**merge)
+
+        # Validate merges
+        for merge in merges:
+            # Check unit ids exist
+            for unit_id in merge.merge_unit_group:
+                if unit_id not in unit_ids:
+                    raise ValueError(f"Merge unit group unit_id {unit_id} is not in the unit list")
+
+            # Check minimum group size
+            if len(merge.merge_unit_group) < 2:
+                raise ValueError("Merge unit groups must have at least 2 elements")
+
+            # Check new unit id not already used
+            if merge.merge_new_unit_id is not None:
+                if merge.merge_new_unit_id in unit_ids:
+                    raise ValueError(f"New unit ID {merge.merge_new_unit_id} is already in the unit list")
+
+        values["merges"] = merges
+        return values
+
+    @classmethod
+    def check_splits(cls, values):
```

Comment:
The structure of split_data should be desribe by mode here.
The split_data type is unclear.